### PR TITLE
Fixed singleselect multi choice widget

### PR DESCRIPTION
**Pull request by sjain28**
Changed single-select multiple choice widget. 

New behavior: 
-Selected text is empty when widget is first displayed. Submitting answer from this view results in null submission
-On dropdown user can select from all of the options specified in formbuilder

In response to http://manage.dimagi.com/default.asp?229724

### CODE REVIEW

File: app/src/org/commcare/views/widgets/SpinnerWidget.java [hunk 1]
```diff
@@ -149,9 +154,16 @@ public View getDropDownView(int position, View convertView, ViewGroup parent) {
...
 
             TextView tv = (TextView)convertView.findViewById(android.R.id.text1);
-            tv.setText(items[position]);
-            tv.setTextSize(textUnit, textSize);
-            tv.setPadding(10, 10, 10, 10); // Are these values OK?
+
+            if(position == 0){
+                tv.setHeight(0); //Hide empty option from dropdown view
+            }else{
+                tv.setHeight(3*(int)textSize+12);
```
Comment (wpride) [inline app/src/org/commcare/views/widgets/SpinnerWidget.java]: This feels weird to me - why are we manually setting the height like this?

**Comment by wpride** [comment]
also think we'd want this in the SpinnerMultiWidget

would be good to get sign off from @dimagi/product

**Comment by sjain28** [comment]
screenshot_20160617-151357
screenshot_20160617-151402
screenshot_20160617-151851
screenshot_20160617-152710

**Comment by amsagoff** [comment]
looks great!

File: app/src/org/commcare/views/widgets/SpinnerWidget.java [hunk 2]
```diff
@@ -149,9 +154,11 @@ public View getDropDownView(int position, View convertView, ViewGroup parent) {
...
 
             TextView tv = (TextView)convertView.findViewById(android.R.id.text1);
-            tv.setText(items[position]);
-            tv.setTextSize(textUnit, textSize);
-            tv.setPadding(10, 10, 10, 10); // Are these values OK?
+
+                tv.setText(items[position]);
+                tv.setTextSize(textUnit, textSize);
+                tv.setPadding(10, 10, 10, 10);
+
```
Comment (phillipm) [inline app/src/org/commcare/views/widgets/SpinnerWidget.java]: Will you remove this indentation change; isn't needed

File: app/src/org/commcare/views/widgets/SpinnerWidget.java [hunk 3]
```diff
@@ -124,18 +123,24 @@ public void setFocus(Context context) {
...
             this.context = context;
             this.textUnit = textUnit;
             this.textSize = textSize;
+
+            //Creates an empty option to be displayed the first time the widget is shown
+            items = new String[objects.length+1];
+            items[0] = "";
+            for(int i = 1; i < items.length; i ++){
+                items[i] = objects[i-1];
+            }
```
Comment (phillipm) [inline app/src/org/commcare/views/widgets/SpinnerWidget.java]: Linter says you should do `System.arraycopy(objects, 0, items, 1, items.length - 1);`. 
Using built-in copy methods is usually faster than doing a manual copying via loops.

**Pull request merged**